src/detect-dataset.c

```diff
@@ -60,6 +61,49 @@ void DetectDatasetRegister (void)
     sigmatch_table[DETECT_DATASET].Free  = DetectDatasetFree;
 }
 
+/*
+    1 match
+    0 no match
+    -1 can't match
+ */
+static int DetectDatajsonBufferMatch(DetectEngineThreadCtx *det_ctx, const DetectDatasetData *sd,
+        const uint8_t *data, const uint32_t data_len)
+{
+    if (data == NULL || data_len == 0)
+        return 0;
+
+    switch (sd->cmd) {
+        case DETECT_DATASET_CMD_ISSET: {
+            // PrintRawDataFp(stdout, data, data_len);
+            DataJsonResultType r = DatajsonLookup(sd->set, data, data_len);
+            SCLogDebug("r found: %d, len: %zu", r.found, r.json.len);
+            if (!r.found)
+                return 0;
+            if (r.json.len > 0) {
+                if ((det_ctx->json_content_len < SIG_JSON_CONTENT_ARRAY_LEN) &&
+                        (r.json.len + strlen(sd->json_key) + 3 < SIG_JSON_CONTENT_ITEM_LEN)) {
+                    snprintf(det_ctx->json_content[det_ctx->json_content_len].json_content,
+                            SIG_JSON_CONTENT_ITEM_LEN, "\"%s\":%s", sd->json_key, r.json.value);
+                    det_ctx->json_content[det_ctx->json_content_len].id = sd->id;
+                    det_ctx->json_content_len++;
+                }
+            }
+            return 1;
+        }
+        case DETECT_DATASET_CMD_ISNOTSET: {
+            // PrintRawDataFp(stdout, data, data_len);
+            DataJsonResultType r = DatajsonLookup(sd->set, data, data_len);
+            SCLogDebug("r found: %d, len: %zu", r.found, r.json.len);
+            if (r.found)
+                return 0;
+            return 1;
+        }
+        default:
+            abort();
```

Comment:
This has been problematic in the past with datasets. ref: https://redmine.openinfosecfoundation.org/issues/7195